File: app/src/main/java/com/naveed/samples/ui/activities/SplashActivity.java
package com.naveed.samples.ui.activities;

import android.os.Bundle;

import com.naveed.samples.R;
import com.naveed.samples.helper.base.SplashBase;

public class SplashActivity extends SplashBase {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_spash);
        startDelay();
    }

    @Override
    protected void onDelayComplete() {
        launchActivity(MainActivity.class);
        finish();
    }
}

File: mvp-rx/src/main/java/com/naveed/samples/ui/activities/SplashActivity.kt
package com.naveed.samples.ui.activities

import android.os.Bundle

import com.naveed.samples.R
import com.naveed.samples.helper.base.SplashBase

class SplashActivity : SplashBase() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_spash)
        startDelay()
    }

    override fun onDelayComplete() {
        launchActivity(MainActivity::class.java)
        finish()
    }
}
